pkgs/by-name/nr/nrfutil/package.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchurl,
+  makeWrapper,
+  libusb1,
+  segger-jlink-headless,
+}:
+
+let
+  source = import ./source.nix;
+  supported = removeAttrs source [ "version" ];
+
+  platform = supported.${stdenvNoCC.system} or (throw "unsupported platform ${stdenvNoCC.system}");
+
+in
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "nrfutil";
+  inherit (source) version;
+
+  src = fetchurl {
+    url = "https://developer.nordicsemi.com/.pc-tools/nrfutil/nrfutil-${platform.name}-${finalAttrs.version}.tar.gz";
+    inherit (platform) hash;
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out
+    mv data/* $out/
+
+    wrapProgram $out/bin/nrfutil \
```

Comment:
Could you use `patchelf --add-needed` instead? That way we wouldn't need any wrapping.

Reply:
I'm not sure it would work too well. `nrfutil` has a plugin-like system that will download external commands that also would need to be patched. The `nrfutil` launcher program doesn't really need `libusb` nor `jlink` itself, it's commands like `nrfutil-device` and possibly others that need them. The external commands get updates asynchronously and are downloaded using an undocumented index that could change at any time, so I think it makes more sense to just manage them through the `nrfutil` tool for now.

Reply:
Ok, no big deal. Generally I consider `LD_LIBRARY_PATH` a code smell, but this makes sense. Thanks!